src/service/mercury/index.ts

```diff
@@ -588,6 +595,14 @@ export class MercuryClient {
       if (!hasIndexerSupport(network)) {
         throw new Error(`network not currently supported: ${network}`);
       }
+      const subs = await this.getAccountSubForPubKey(pubKey, network);
+      const hasSubForPublicKey = subs.some(
+        (sub: { publickey: string }) => sub.publickey === pubKey
+      );
+      if (!hasSubForPublicKey) {
+        throw new Error(ERROR.MISSING_SUB_FOR_PUBKEY);
+      }
```

Comment:
since these are the same check twice - should they be abstracted into a helper? I guess the tricky thing is you want the `throw` to still happen in this function so it gets caught in `getAccountHistory`

Reply:
Sure they can be, moved that to a helper in 4324b208212d9dc73462353dd583353f789d160b